3-enrich/stream-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.stream/PiiEmitSpec.scala

```diff
@@ -0,0 +1,215 @@
+      val parsedConfig     = ConfigFactory.parseString(configInstance).resolve()
+      val configObject     = Try { loadConfigOrThrow[EnrichConfig](parsedConfig.getConfig("enrich")) }
+      configObject aka "enrichment config loading" must not beAFailedTry
+      val app = Future {
+        EnrichApp.run(configObject.get, SpecHelpers.resolver, SpecHelpers.enrichmentRegistry, None)
+      }
+      // Input
+      val inputGood = List(
+        decode(PagePingWithContextSpec.raw),
+        decode(PageViewWithContextSpec.raw),
+        decode(StructEventSpec.raw),
+        decode(StructEventWithContextSpec.raw),
+        decode(TransactionItemSpec.raw),
+        decode(TransactionSpec.raw)
+      )
+      val (expectedGood, expectedBad, expectedPii) = (inputGood.size, 0, inputGood.size)
+
+      import ExecutionContext.Implicits.global
+
+      val producer = Future {
+        val testKafkaPropertiesProducer = {
+          val props = new Properties()
+          props.put("bootstrap.servers", kafkaHost)
+          props.put("client.id", "producer-george")
+          props.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer")
+          props.put("value.serializer", "org.apache.kafka.common.serialization.ByteArraySerializer")
+          props
+        }
+        val testProducer = new KafkaProducer[String, Array[Byte]](testKafkaPropertiesProducer)
+
+        val events = inputGood
+        events.foreach { r =>
+          testProducer.send(new ProducerRecord(testGoodIn, "key", r))
+        }
+        testProducer.flush
+        testProducer.close
+      }
+      val producerTimeoutSec = 5
+      val inputProduced      = Try { Await.result(producer, Duration(s"$producerTimeoutSec sec")) }
+      inputProduced aka "sending input" must beSuccessfulTry
+
+      private def getRecords(cr: ConsumerRecords[String, String]): List[String] =
+        cr.map(_.value).toList
+
+      private def getConsumer(topic: String, expectedRecords: Int, timeoutSec: Long) = Future {
+        val started = System.currentTimeMillis
+        val testKafkaPropertiesConsumer = {
+          val props = new Properties()
+          props.put("bootstrap.servers", kafkaHost)
+          props.put("key.deserializer", "org.apache.kafka.common.serialization.StringDeserializer")
+          props
+            .put("value.deserializer", "org.apache.kafka.common.serialization.StringDeserializer")
+          props.put("group.id", s"consumer-$topic")
+          props
+        }
+        val testConsumerPii = new KafkaConsumer[String, String](testKafkaPropertiesConsumer)
+        testConsumerPii.subscribe(List(topic))
+        var records = getRecords(testConsumerPii.poll(100))
+        while (((System.currentTimeMillis - started) / 1000 < timeoutSec - 1) && records.size < expectedRecords) {
+          records = records ++ getRecords(testConsumerPii.poll(100))
+        }
+        records
+      }
```

Comment:
Are you referring to the properties above? (`key.deserializer` etc.)

Reply:
I mean `producer`, `getConsumer`, `getRecords`, etc

Reply:
As I don't know of the use case you have in mind (another test?) can you please tell me where exactly (package? SpecHelpers?) they need be extracted? Alternatively if you have an additional use case, I can refactor and add the new test to this release (lets create a ticket in this milestone for that). I think using tests like that more could be great as they will improve coverage.

Reply:
>I think using tests like that more could be great as they will improve coverage.

That's exactly what I meant. So what would be easiest would be to extract all this into a trait so that a spec can just extend it and have kafka integration.

Reply:
👍